### PR TITLE
fixing changes in 39

### PR DESCRIPTION
**Pull request by prateekbh**
- fixing dom for drawers
- fixing doc's site
- fix icon button
- fixing drawers docs

### CODE REVIEW

**Comment by artipost** [comment]
Failed pictures (28): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/5c52af6e16965a25a62bab14513ecafc/raw/d22d8ea58b0293d6cfae13724bec1423c3320095/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [427472684](https://travis-ci.org/prateekbh/preact-material-components/builds/427472684), for commit [1af92b8](https://github.com/prateekbh/preact-material-components/pull/994/commits/1af92b8f1aa7c37a931478913627740449e0f60e).

**Comment by artipost** [comment]
Failed pictures (28): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/cef25b06365ad6a7b55dd574e2a4bb2c/raw/aa5a6245a970bf23480a032c8f725340d6d24448/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [429391192](https://travis-ci.org/prateekbh/preact-material-components/builds/429391192), for commit [4401d4b](https://github.com/prateekbh/preact-material-components/pull/994/commits/4401d4bccf3a9a09dc40e72751f36a89dfc12eca).

**Comment by cromefire** [comment]
Will take a look at this later this day

**Comment by cromefire** [comment]
Please take a look at the the Travis warnings and errors, there are missing peer dependencies and tslint errors. Pleas fix them or disable the rules (the alphabetical order object literal one is really annoying and has no fixer)

**Comment by cromefire** [comment]
The branch test doesn't want to run:
```
./scripts/travis.sh: line 55: [: ==: unary operator expected
Nothing has changed, no need to run tests.
```

File: docs/src/routes/icon-button/index.js [hunk 1]
```diff
@@ -18,38 +20,21 @@ export default class IconButtonPage extends Component {
     this.propsTable = [
       {
         component: 'IconButton',
+        props: []
+      },
+      {
+        component: 'Button',
```
Comment (cromefire) [inline docs/src/routes/icon-button/index.js]: Are you sure that this is right? I thought that has to be IconButton and IconButton.Icon

File: ts/Drawer/index.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
-import {MDCPersistentDrawer} from '@material/drawer/persistent';
-import {MDCTemporaryDrawer} from '@material/drawer/temporary';
+// @ts-ignore
```
Comment (cromefire) [inline ts/Drawer/index.tsx]: Are the updates to the TypeScript definitions?

File: ts/IconButton/index.tsx [hunk 3]
```diff
@@ -1,50 +1,53 @@
...
+    const ButtonElement = props.element || 'i';
+    return (
+      <ButtonElement class="material-icons" {...props}>
+        {props.children}
+      </ButtonElement>
+    );
+  }
+}
+
+class IconButton extends MaterialComponent<IIconButtonProps, IIconButtonState> {
```
Comment (cromefire) [inline ts/IconButton/index.tsx]: Export this

File: ts/IconButton/index.tsx [hunk 4]
```diff
@@ -1,50 +1,53 @@
...
-> {
-  public static readonly Icon = Icon;
+export interface IButtonProps {
+  on?: boolean;
+  element?: boolean;
+}
+
+export interface IButtonState {}
+
+class Button extends MaterialComponent<IButtonProps, IButtonState> {
```
Comment (cromefire) [inline ts/IconButton/index.tsx]: Shouldn't that be called `Icon`? And also export this

File: docs/src/routes/icon-button/index.js (hunk 1 above)
Comment (prateekbh) [reply docs/src/routes/icon-button/index.js]: its kinda upto us to name it

File: ts/Drawer/index.tsx (hunk 2 above)
Comment (prateekbh) [reply ts/Drawer/index.tsx]: @type did not update after 0.35 i am in favor of removing it

File: ts/IconButton/index.tsx (hunk 4 above)
Comment (prateekbh) [reply ts/IconButton/index.tsx]: actually yes

File: ts/Drawer/index.tsx (hunk 2 above)
Comment (cromefire) [reply ts/Drawer/index.tsx]: Yes I think until it updates (if it ever does) we should remove it

File: docs/src/routes/icon-button/index.js (hunk 1 above)
Comment (cromefire) [reply docs/src/routes/icon-button/index.js]: Well naming an icon button isn't quite intuitive it's the same one as in my last comment

Comment (cromefire) [reply docs/src/routes/icon-button/index.js]: https://github.com/prateekbh/preact-material-components/blob/96aba2428a7ee18e3dd474770341c0b71f214171/ts/IconButton/index.tsx#L32

It's actually named half Icon, half button in code

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/79cf4c87bb2f2b1169d28b1208c5aee6/raw/07fe26c3c6276b208c26797c29447926ae7de2c5/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [429643395](https://travis-ci.org/prateekbh/preact-material-components/builds/429643395), for commit [13f0145](https://github.com/prateekbh/preact-material-components/pull/994/commits/13f0145049c3f4a73086efff7c28a7aa73dae710).

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/4219ed14adf6a184aaeb0b4407f925c1/raw/8e56f9d441515acb13d623f7aeab78141a094452/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [429764227](https://travis-ci.org/prateekbh/preact-material-components/builds/429764227), for commit [4874784](https://github.com/prateekbh/preact-material-components/pull/994/commits/4874784e0e2ff847082e1b121624602830df7a20).

**Comment by prateekbh** [comment]
@cromefire can you please review?

**Comment by cromefire** [comment]
If tomorrow is fine

File: docs/src/routes/icon-button/index.js (hunk 1 above)
Comment (cromefire) [reply docs/src/routes/icon-button/index.js]: This is still to change if you don't want to sell the users an icon as a button

File: ts/Drawer/index.tsx (hunk 2 above)
Comment (cromefire) [reply ts/Drawer/index.tsx]: Then please remove it and then remove that `@ts-ignore`

**Comment by prateekbh** [comment]
@cromefire whats wrong with `TopAppBar` I just replaced the deprecated component with the new one and hence the color change. Do you want me to fix the background color?

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/6cdd72606883d22980af3eba81f5299d/raw/aad37b87eda8ab8ebdf08a7de26657117655107c/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [431111984](https://travis-ci.org/prateekbh/preact-material-components/builds/431111984), for commit [c86c5be](https://github.com/prateekbh/preact-material-components/pull/994/commits/c86c5be395e94717f52a52badb4aeaa7f38e7e00).

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/c6cb28aa0f967f75805ccfbb4b6a46a1/raw/42ad2dfc07773588d4b8a0751d616e9414a4a351/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [431117731](https://travis-ci.org/prateekbh/preact-material-components/builds/431117731), for commit [4ccaaf5](https://github.com/prateekbh/preact-material-components/pull/994/commits/4ccaaf5e4b7b6d8db347e83789e8377e608cd5fc).

**Comment by cromefire** [comment]
@prateekbh oh then please annotate that it is deprecated because the site in the docs is just blank

**Comment by prateekbh** [comment]
@cromefire i got what u were saying.. fixed it already

**Comment by prateekbh** [comment]
@cromefire done

**Comment by cromefire** [comment]
You can merge it

**Pull request merged**